File: CS/WinWebSolution.Web/ApplicationCode/WebApplicationE1344.cs
using DevExpress.ExpressApp;
using WinWebSolution.Module;

namespace WinWebSolution.Web {
    partial class WinWebSolutionAspNetApplication : ISupportChangeDatabaseAtRuntime {
        public void ChangeTo(string newConnectionString) {
            IXpoDataStoreProvider dataStoreProvider = new ConnectionStringDataStoreProvider(newConnectionString);
            Setup(new ExpressApplicationSetupParameters(ApplicationName, CreateDefaultObjectSpaceProvider(dataStoreProvider), ControllersManager, DetailViewItemsFactory, Model, Modules));
            System.Web.HttpContext.Current.Response.Redirect("Default.aspx");
        }
    }
}

File: VB/WinWebSolution.Web/ApplicationCode/WebApplicationE1344.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.ExpressApp
Imports WinWebSolution.Module

Namespace WinWebSolution.Web
    Partial Public Class WinWebSolutionAspNetApplication
        Implements ISupportChangeDatabaseAtRuntime
        Public Sub ChangeTo(ByVal newConnectionString As String) Implements ISupportChangeDatabaseAtRuntime.ChangeTo
            Dim dataStoreProvider As IXpoDataStoreProvider = New ConnectionStringDataStoreProvider(newConnectionString)
            Setup(New ExpressApplicationSetupParameters(ApplicationName, CreateDefaultObjectSpaceProvider(dataStoreProvider), ControllersManager, DetailViewItemsFactory, Model, Modules))
            System.Web.HttpContext.Current.Response.Redirect("Default.aspx")
        End Sub
    End Class
End Namespace
